### ISSUE TITLE
Cannot parse AssemblyInfo.cs files generated by MonoDevelop

### ISSUE THREAD

**Issue by pvwichen** [cake-build/cake#943]
### What You Are Seeing?
When using the alias `ParseAssemblyInfo`, it returns null values if the `AssemblyInfo.cs` file was generated by MonoDevelop even though the file appears to be valid.

MonoDevelop is adding an extra space between the attribute name and the opening parenthesis.
`[assembly: AssemblyTitle ("AssemblyTitle")]`

### What is Expected?
When using `ParseAssemblyInfo`, its regular expressions should allow for whitespace between the attribute name and opening parenthesis.

I can create a pull request if accepted.

### What version of Cake are you using?
Cake 0.12.0

### Are you running on a 32 or 64 bit system?
64-bit

### What environment are you running on?  Windows? Linux? Mac?
Windows, Linux, Mac

### Are you running on a CI Server?  If so, which one?
Jenkins, TeamCity

### How Did You Get This To Happen? (Steps to Reproduce)

Use `ParseAssemblyInfo` to parse an AssemblyInfo.cs file containing:
```
using System.Reflection; 
using System.Runtime.CompilerServices; 
 
// Information about this assembly is defined by the following attributes. 
// Change them to the values specific to your project. 
 
[assembly: AssemblyTitle ("AssemblyTitle")] 
[assembly: AssemblyDescription ("AssemblyDescription")] 
[assembly: AssemblyConfiguration ("")] 
[assembly: AssemblyCompany ("")] 
[assembly: AssemblyProduct ("")] 
[assembly: AssemblyCopyright ("Copyright")] 
[assembly: AssemblyTrademark ("")] 
[assembly: AssemblyCulture ("")] 
 
// The assembly version has the format "{Major}.{Minor}.{Build}.{Revision}". 
// The form "{Major}.{Minor}.*" will automatically update the build and revision, 
// and "{Major}.{Minor}.{Build}.*" will update just the revision. 
 
[assembly: AssemblyVersion ("1.0.*")] 
 
// The following attributes are used to specify the signing key for the assembly, 
// if desired. See the Mono documentation for more information about signing. 
 
//[assembly: AssemblyDelaySign(false)] 
//[assembly: AssemblyKeyFile("")] 
```

`AssemblyDescription` will be null.

**Comment by devlead** [comment]
@pvwichen found what's casuing it, file above adds an space between attribute and parentheses which apparently breaks the parser.
VisualStudio doesn't add these spaces, so there seems to be a slight indiscretion between MonoDevelop and VS.

I created a gist with a small test case [here](https://gist.github.com/devlead/a5a43a42d4ea8de95a8afbeb79c468f9) as reference when sorting issue.

Basically:
```csharp
using System.Reflection;
using System.Runtime.CompilerServices;

// Information about this assembly is defined by the following attributes.
// Change them to the values specific to your project.

[assembly: AssemblyTitle ("AssemblyTitle")]
[assembly: AssemblyDescription ("AssemblyDescription")]
[assembly: AssemblyConfiguration ("Configuration")]
[assembly: AssemblyCompany ("Company")]
[assembly: AssemblyProduct ("Product")]
[assembly: AssemblyCopyright ("Copyright")]
[assembly: AssemblyTrademark ("Trademark")]
[assembly: AssemblyCulture ("en-US")]

// The assembly version has the format "{Major}.{Minor}.{Build}.{Revision}".
// The form "{Major}.{Minor}.*" will automatically update the build and revision,
// and "{Major}.{Minor}.{Build}.*" will update just the revision.

[assembly: AssemblyVersion ("1.0.*")]

// The following attributes are used to specify the signing key for the assembly,
// if desired. See the Mono documentation for more information about signing.

//[assembly: AssemblyDelaySign(false)]
//[assembly: AssemblyKeyFile("")]
```
will fail but this will succeed:
```csharp
using System.Reflection;
using System.Runtime.CompilerServices;

// Information about this assembly is defined by the following attributes.
// Change them to the values specific to your project.

[assembly: AssemblyTitle("AssemblyTitle")]
[assembly: AssemblyDescription("AssemblyDescription")]
[assembly: AssemblyConfiguration("Configuration")]
[assembly: AssemblyCompany("Company")]
[assembly: AssemblyProduct("Product")]
[assembly: AssemblyCopyright("Copyright")]
[assembly: AssemblyTrademark("Trademark")]
[assembly: AssemblyCulture("en-US")]

// The assembly version has the format "{Major}.{Minor}.{Build}.{Revision}".
// The form "{Major}.{Minor}.*" will automatically update the build and revision,
// and "{Major}.{Minor}.{Build}.*" will update just the revision.

[assembly: AssemblyVersion("1.0.*")]

// The following attributes are used to specify the signing key for the assembly,
// if desired. See the Mono documentation for more information about signing.

//[assembly: AssemblyDelaySign(false)]
//[assembly: AssemblyKeyFile("")]
```

Test case will output
```
Fail:
{
  "ClsCompliant": false,
  "Company": "",
  "ComVisible": false,
  "Configuration": "",
  "Copyright": "",
  "Description": "",
  "AssemblyFileVersion": "1.0.0.0",
  "Guid": "",
  "AssemblyInformationalVersion": "1.0.0.0",
  "Product": "",
  "Title": "",
  "Trademark": "",
  "AssemblyVersion": "1.0.0.0",
  "InternalsVisibleTo": []
}
Success:
{
  "ClsCompliant": false,
  "Company": "Company",
  "ComVisible": false,
  "Configuration": "Configuration",
  "Copyright": "Copyright",
  "Description": "AssemblyDescription",
  "AssemblyFileVersion": "1.0.0.0",
  "Guid": "",
  "AssemblyInformationalVersion": "1.0.0.0",
  "Product": "Product",
  "Title": "AssemblyTitle",
  "Trademark": "Trademark",
  "AssemblyVersion": "1.0.*",
  "InternalsVisibleTo": []
}
```

**Comment by gep13** [comment]
@pvwichen this was a simple change to the regular expression that is used to extract the data.  I have created a PR to correct this, and to add some more tests around how it is parsed.

**Comment by gep13** [comment]
@pvwichen this should be fixed in the next release of Cake, 0.13.0.
